microservices/Keli-Katheya-User-Registration/src/main/java/com/kelikatheya/microservices/userRegistration/services/impls/UserServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.kelikatheya.microservices.userRegistration.services.impls;
+
+import com.kelikatheya.microservices.userRegistration.entity.UserEntity;
+import com.kelikatheya.microservices.userRegistration.entity.UserRepository;
+import com.kelikatheya.microservices.userRegistration.model.UserModel;
+import com.kelikatheya.microservices.userRegistration.requestModels.UserRequest;
+import com.kelikatheya.microservices.userRegistration.services.UserService;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.convention.MatchingStrategies;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Override
+    public UserModel registerUser(UserRequest userRequest) {
+
+        ModelMapper modelMapper = new ModelMapper();
+        modelMapper.getConfiguration().setMatchingStrategy(MatchingStrategies.STRICT);
+        //map fields from userRequest to userEntity
+        UserEntity userEntity = modelMapper.map(userRequest , UserEntity.class);
```

Comment:
Shouldn't we handle mapping exceptions?

Reply:
No need I believe, since both the class will have same set of fields, if field doesn't matches it'll ignore

Reply:
Okay